pilot/cmd/pilot-agent/status/server.go

```diff
@@ -144,10 +146,10 @@ type Server struct {
 	config                Options
 	http                  *http.Client
 	enableProfiling       bool
-	registry              *prometheus.Registry
+	registry              prometheus.Gatherer
 }
 
-func initializeMonitoring() (*prometheus.Registry, error) {
+func initializeMonitoring() (prometheus.Gatherer, error) {
```

Comment:
If we're doing this - shouldn't agent also use otel sdk ? Not in this PR, of course - but a TODO and comment explaining why prometheus is used directly would help other developers.